daml/Marketplace/Issuance.daml

```diff
@@ -33,6 +45,34 @@ template Service
     maintainer key._1
 
     controller client can
+      nonconsuming RequestOrigination: ContractId OriginationRequest
+        with
+          assetId: Id
+          description: Text
+          issuer: Party
+          claims: Claims
+          safekeepingAccountId: Id
+        do
+          create OriginationRequest with ..
+
+      nonconsuming RequestBinaryCallOptionOrigination: ContractId OriginationRequest
+        with
+          assetId: Id
+          issuer: Party
+          safekeepingAccountId: Id
+          expiry: Date
+          strike: Decimal
+          underlyingId: Id
+          currencyId: Id
+        do
+          -- TODO: refactor into a more composable form and add to contingent-claims lib
+          let binary : Decimal -> Text -> Id -> Claim O.Observation Date Id
+              binary strike spot ccy = cond (O.pure strike O.<= O.observe spot) (one currencyId) Zero
```

Comment:
this would look neater without the `O.` qualifier

Reply:
Yes, the problem is that it would clash with prelude. If you look in `ContingentClaims/FinancialClaim.daml` you'll see that you can get around this by using `{-# NoImplicitPrelude #-}` but I hesitate to put that in here.

The solution is really to have this in a different file e.g. `FinancialClaim.daml`, as these are the lowest-level combinators, that I wouldn't expect a user to access directly. They would probably be using higher level combinators like `zcb` or `european` (see also my comment in the code :grin: )

Reply:
btw `O.<=` is `(<=) : f a -> f a -> f Bool` rather than prelude's `(<=) a -> a -> Bool`, hence the difference.

Reply:
I think you can explicitly import Prelude with an exclusion list even. Anyway, not an issue for now